client/api_provider.go

```diff
@@ -281,3 +286,17 @@ func (a *apiProvider) GetJSONFeature(ctx context.Context, key string, namespace
 	}
 	return nil
 }
+
+type ProviderOption interface {
+	apply(provider *apiProvider)
+}
+
+// Used to override the default Lekko backend API URL.
+// Only applicable to API provider.
+type URLOption struct {
+	URL string
+}
+
+func (uo *URLOption) apply(provider *apiProvider) {
+	provider.lekkoClient = clientv1beta1connect.NewConfigurationServiceClient(http.DefaultClient, uo.URL)
```

Comment:
Allowing custom options to directly change internal sounds a bit hack-y. What about having "options" struct that custom options can change and then using it during initialization?

Reply:
If I understand correctly, I think it's fine and behavior I've seen similarly in other libraries - the pattern is that options that implement the interface do have direct access to modify the internals of what they're options for.